### PR TITLE
ref 635294: server update schedules

### PR DESCRIPTION
**Pull request by aeter**
Adds terraform datasources and resources for the STACKIT `server update` API.

Testing Done:
* ran `make fmt`, `make lint`, `make generate-docs`, `make test`
* tested locally by creating, updating, importing, deleting update schedules.

### CODE REVIEW

File: stackit/internal/services/serverupdate/schedule/resource.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+	tflog.Debug(ctx, "Enabling server update service")
+	err := client.EnableService(ctx, projectId, serverId).EnableServicePayload(enableServicePayload).Execute()
+	if err != nil {
+		if strings.Contains(err.Error(), "Tried to activate already active service") {
+			tflog.Debug(ctx, "Service for server update already enabled")
+			return nil
+		}
+		return fmt.Errorf("enable server update service: %w", err)
+	}
```
Comment (joaopalet) [inline stackit/internal/services/serverupdate/schedule/resource.go:423]: Is this the only way to do it? Would be nice if the endpoint was idempotent or if it returned a specific status code. Othwrwise we have to depend on the error message

File: stackit/internal/services/serverupdate/schedule/schedules_datasource.go
Comment (joaopalet) [inline stackit/internal/services/serverupdate/schedule/schedules_datasource.go:1]: :+1:

File: stackit/internal/services/serverupdate/serverupdate_acc_test.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+					server_id  = "%s"
+					name  = "%s"
+ 				 	rrule = "%s"
+                    enabled = true
+                    update_properties = {
+                        name = "%s"
+                        retention_period = %d
+                        volume_ids = null
+                    }
+				}
```
Comment (joaopalet) [inline stackit/internal/services/serverupdate/serverupdate_acc_test.go:44]: The `update_properties` field is not specified in the resource schema, right?

Comment (joaopalet) [reply stackit/internal/services/serverupdate/serverupdate_acc_test.go:44]: Also it seems to be missing a required `maintenance_window` field

Comment (joaopalet) [reply stackit/internal/services/serverupdate/serverupdate_acc_test.go:44]: Does this test run and pass?

File: stackit/internal/services/serverupdate/serverupdate_acc_test.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+				`,
+		testutil.ServerUpdateProviderConfig(),
+		serverUpdateScheduleResource["project_id"],
+		serverUpdateScheduleResource["server_id"],
+		serverUpdateScheduleResource["update_schedule_name"],
+		serverUpdateScheduleResource["rrule"],
+		serverUpdateScheduleResource["update_name"],
+	)
+}
```
Comment (joaopalet) [inline stackit/internal/services/serverupdate/serverupdate_acc_test.go:80]: Is this extra function needed? Couldn't we use the one above with extra argument(s) for the fields which are updated?

File: docs/resources/resourcemanager_project.md
Comment (joaopalet) [inline docs/resources/resourcemanager_project.md:1]: Could you please revert the changes to this file (and the one for the data source as well)?

Comment (joaopalet) [reply docs/resources/resourcemanager_project.md:1]: We have deprecated a field that we don't want to show in the documentation anymore

File: stackit/internal/services/serverupdate/schedule/resource.go (hunk 1 above)
Comment (aeter) [reply stackit/internal/services/serverupdate/schedule/resource.go:423]: Yes, currently it is. Sorry.

File: stackit/internal/services/serverupdate/serverupdate_acc_test.go (hunk 2 above)
Comment (aeter) [reply stackit/internal/services/serverupdate/serverupdate_acc_test.go]: Yes, the test was passing...strange. Updated the test code, ran it explicitly. Thanks

**Review by joaopalet — APPROVED** [review]
